packages/loaders/src/elements/Dots.tsx

```diff
@@ -39,14 +40,23 @@ export interface IDotsProps extends React.HTMLAttributes<SVGSVGElement> {
 }
 
 const Dots: React.FC<IDotsProps> = ({ size, color, duration, delayMS, ...other }) => {
-  const noAnimatedSVGSupport = useCSSSVGAnimation();
-  const { delayComplete } = useSchedule({ duration, delayMS, loop: noAnimatedSVGSupport });
+  const theme = useContext(ThemeContext);
+  const environment = useDocument(theme);
+  const canTransformSVG = useRef<boolean | null>(null);
+
+  if (environment && canTransformSVG.current === null) {
+    const svg = environment.createElementNS('http://www.w3.org/2000/svg', 'svg');
+
+    canTransformSVG.current = 'transform' in svg;
+  }
```

Comment:
For this to support SSR (and the website) wouldn't this need to be in a `useLayoutEffect` to time the calculation correctly?

Reply:
I tested the updated `Dot` component on the Garden website (and vanilla Next.js) app and it seems to work as expected.

Should SSR avoid `useLayoutEffect` completely? From the [docs](https://reactjs.org/docs/hooks-reference.html#uselayouteffect):

> If you use server rendering, keep in mind that neither useLayoutEffect nor useEffect can run until the JavaScript is downloaded. This is why React warns when a server-rendered component contains useLayoutEffect. To fix this, either move that logic to useEffect (if it isn’t necessary for the first render), or delay showing that component until after the client renders (if the HTML looks broken until useLayoutEffect runs).

Reply:
Oh nice, haven't seen this warning before. I would revise my comment to just suggest `useEffect` then.

Did you run a full production build for the Website? [useDocument should return undefined originally](https://github.com/zendeskgarden/react-components/blob/main/packages/theming/src/utils/useDocument.ts)

Reply:
I only ran development builds of the website and Next.js app, but I can run a full production build for a sanity check.

It seems that wrapping this code in a `useEffect` is beneficial to prevent unnecessary calculations of creating an SVG on additional renders. But I think the `null` check already achieves this. It's why I opted for storing the boolean check into a `useRef`.

Reply:
LGTM!